cg/services/analysis_starter/configurator/implementations/balsamic.py

```diff
@@ -0,0 +1,259 @@
+            artefact_snv_observations=self.loqusdb_artefact_snv,
+            balsamic_binary=self.balsamic_binary,
+            balsamic_cache=self.cache_dir,
+            cadd_annotations=self.cadd_path,
+            cancer_germline_snv_observations=self.loqusdb_cancer_germline_snv,
+            cancer_germline_sv_observations=self.loqusdb_cancer_germline_sv,
+            cancer_somatic_snv_observations=self.loqusdb_cancer_somatic_snv,
+            cancer_somatic_sv_observations=self.loqusdb_cancer_somatic_sv,
+            case_id=case.internal_id,
+            clinical_snv_observations=self.loqusdb_clinical_snv,
+            clinical_sv_observations=self.loqusdb_clinical_sv,
+            conda_binary=self.conda_binary,
+            fastq_path=Path(self.root_dir, case.internal_id, "fastq"),
+            gender=patient_sex,
+            genome_version=GenomeVersion.HG19,
+            gnomad_min_af5=self.gnomad_af5_path,
+            normal_sample_name=self._get_normal_sample_id(case),
+            panel_bed=bed_file,
+            pon_cnn=self._get_pon(bed_file),
+            exome=self._all_samples_are_exome(case),
+            sentieon_install_dir=self.sentieon_licence_path,
+            sentieon_license=self.sentieon_licence_server,
+            soft_filter_normal=bool(self._get_normal_sample_id(case)),
+            swegen_snv=self.swegen_snv,
+            swegen_sv=self.swegen_sv,
+            tumor_sample_name=self._get_tumor_sample_id(case),
+        )
+
+    @staticmethod
+    def _all_samples_are_wgs(case: Case) -> bool:
+        """Check if all samples in the case are WGS."""
+        return all(
+            sample.application_version.application.prep_category
+            == SeqLibraryPrepCategory.WHOLE_GENOME_SEQUENCING
+            for sample in case.samples
+        )
+
+    @staticmethod
+    def _all_samples_are_exome(case: Case) -> bool:
+        """Check if all samples in the case are exome."""
+        return all(
+            sample.application_version.application.prep_category
+            == SeqLibraryPrepCategory.WHOLE_EXOME_SEQUENCING
+            for sample in case.samples
+        )
+
+    @staticmethod
+    def _get_patient_sex(case) -> SexOptions:
+        sample_sex: set[SexOptions] = {sample.sex for sample in case.samples}
+        if len(sample_sex) == 1:
+            return sample_sex.pop()
+        else:
+            raise BalsamicInconsistentSexError(
+                f"Case {case.internal_id} contains samples of differing sex"
+            )
+
+    @staticmethod
+    def _get_normal_sample_id(case) -> str | None:
+        for sample in case.samples:
+            if not sample.is_tumour:
+                return sample.internal_id
+
+    @staticmethod
+    def _get_tumor_sample_id(case) -> str:
+        for sample in case.samples:
+            if sample.is_tumour:
+                return sample.internal_id
+        raise BalsamicMissingTumorError(f"Case {case.internal_id} does not contain a tumor sample")
```

Comment:
same for this exception, it should not happen

Reply:
I think it should be kept